core/src/main/scala/org/ensime/core/AnalyzerManager.scala

```diff
@@ -45,7 +45,7 @@ class AnalyzerManager(
       case Some(moduleId) =>
         f(req, moduleId)
       case None =>
-        sauron ! EnsimeServerError(s"Couldn't find the project for ${fileInfo.file}")
+        sauron forward req
```

Comment:
and if you put the logic in `getOrSpawnNew` then you don't even need to change most of the existing logic.